openvdb/openvdb/points/PointRasterizeSDF.h

```diff
@@ -0,0 +1,1701 @@
+template <typename RadiusTreeT,
+    typename MaskTreeT,
+    typename InterrupterT = util::NullInterrupter>
+struct VariableSurfaceMaskOp
+    : public SurfaceMaskOp<MaskTreeT, InterrupterT>
+{
+    using BaseT = SurfaceMaskOp<MaskTreeT, InterrupterT>;
+    using LeafManagerT =
+        tree::LeafManager<const points::PointDataTree>;
+
+    VariableSurfaceMaskOp(const math::Transform& pointsTransform,
+                         const math::Transform& surfaceTransform,
+                         const RadiusTreeT& min,
+                         const RadiusTreeT& max,
+                         const Real scale,
+                         const Real halfband,
+                         InterrupterT* interrupter = nullptr)
+        : BaseT(pointsTransform, surfaceTransform, interrupter)
+        , mMin(min), mMax(max), mScale(scale), mHalfband(halfband) {}
+
+    VariableSurfaceMaskOp(const VariableSurfaceMaskOp& other, tbb::split)
+        : BaseT(other), mMin(other.mMin), mMax(other.mMax)
+        , mScale(other.mScale), mHalfband(other.mHalfband) {}
+
+    void operator()(const typename LeafManagerT::LeafRange& range)
+    {
+        if (this->interrupted()) return;
+        const tree::ValueAccessor<const RadiusTreeT> maxacc(mMax);
+        for (auto leafIter = range.begin(); leafIter; ++leafIter) {
+            const int max = this->maxDist(maxacc.getValue(leafIter->origin()));
+            this->fill(*leafIter, max, true);
+        }
+        const tree::ValueAccessor<const RadiusTreeT> minacc(mMin);
+        for (auto leafIter = range.begin(); leafIter; ++leafIter) {
+            const int min = this->minDist(minacc.getValue(leafIter->origin()));
+            if (min <= 0) continue;
+            this->fill(*leafIter, min, false);
+        }
+    }
+
+private:
+    inline int maxDist(const typename RadiusTreeT::ValueType& max) const
+    {
+        // max radius in index space
+        const Real v = (Real(max) * mScale) + mHalfband;
+        int d = int(math::RoundUp(v)); // next voxel
+        d += this->mVoxelOffset; // final offset
+        return d;
+    }
+
+    inline int minDist(const typename RadiusTreeT::ValueType& min) const
+    {
+        // min radius in index space
+        Real v = math::Max(0.0, (Real(min) * mScale) - mHalfband);
+        v = math::Max(0.0, Real(v - this->mVoxelOffset)); // final offset
+        // calculate the min interior cube area of activity. this is the side
+        // of the largest possible cube that fits into the radius "v":
+        //   d = 2r -> 2r = 3x^2 -> x = 2r / sqrt(3)
+        // finally half it to the new radius
+        v = ((2.0 * v) / std::sqrt(3.0)) / 2.0;
+        return static_cast<int>(v); // round down (v is always >= 0)
+    }
+
+private:
+    const RadiusTreeT& mMin;
+    const RadiusTreeT& mMax;
+    const Real mScale, mHalfband;
+};
+
+template <typename SdfT, typename InterrupterT, typename PointDataGridT>
+inline typename SdfT::Ptr
```

Comment:
Template functions should not be inline, this does nothing except prevent explicit instantiation from being used.